### PR TITLE
Add JsonRPC protocol contract

### PR DESCRIPTION
**Pull request by ferranbt**
This PR introduces a JsonRPC protocol contract to call the Ethereum JsonRPC endpoints. This PR implements the get nonce endpoint.

### CODE REVIEW

File: src/protocols/JsonRPC.sol [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.13;
+
+import "src/suavelib/Suave.sol";
+import "solady/src/utils/JSONParserLib.sol";
+import "solady/src/utils/LibString.sol";
+
+contract JsonRPC {
```
Comment (dmarzzz) [inline src/protocols/JsonRPC.sol:8]: we should call this EthJsonRPC I think?

**Review by dmarzzz — CHANGES_REQUESTED** [review]

File: src/protocols/JsonRPC.sol (hunk 1 above)
Comment (ferranbt) [reply src/protocols/JsonRPC.sol]: Done

File: README.md [hunk 2]
```diff
@@ -84,6 +84,23 @@ contract Example {
...
+Helper library to interact with the Ethereum JsonRPC protocol.
+
+#### Example usage
+
+```solidity
+import "suave-std/protocols/JsonRPC.sol";
+
+contract Example {
+    function example() {
+        JsonRPC jsonrpc = new JsonRPC("http://...");
```
Comment (metachris) [inline README.md:98]:
```suggestion
        JsonRPC jsonrpc = new EthJsonRPC("http://...");
```

Comment (ferranbt) [reply README.md]: Fixed

**Review by dmarzzz — APPROVED** [review]

**Pull request merged**